cg/store/crud/read.py

```diff
@@ -1562,3 +1562,16 @@ def get_case_ids_for_samples(self, sample_ids: list[int]) -> list[str]:
         for sample_id in sample_ids:
             case_ids.extend(self.get_case_ids_with_sample(sample_id))
         return list(set(case_ids))
+
+    def sample_exists_with_different_sex(
+        self,
+        customer_internal_id: str,
+        subject_id: str,
+        sex: str,
+    ) -> bool:
+        samples: list[Sample] = self.get_samples_by_customer_and_subject_id(
+            customer_internal_id=customer_internal_id,
+            subject_id=subject_id,
+        )
+        existing_sex: set[str] = {sample.sex for sample in samples}
```

Comment:
In essence, a subject_id corresponds to one individual. Hence, one could argue that if this set contains anything other than one element, the data is off. A bit uncertain if it can be a mix of a known sex and unknown though.

Reply:
So this check is not perfect. If the sex in the db is "unkown" and they provide a "female", then the check should not fail.